src/development/packages-and-plugins/developing-packages.md

```diff
@@ -614,6 +614,16 @@ a section in [Supporting the new Android plugins APIs][].
 If you want to develop a package that calls into native APIs using
 Dart's FFI, you need to develop an FFI plugin package.
 
+Both FFI plugin packages and (non-FFI) plugin packages support
+bundling native code, but FFI plugin packages do not support
+method channels and do include method channel registration code.
+If you want to implement a plugin that uses both method channels
+and FFI, use a (non-FFI) plugin. You can chose per platform to
```

Comment:
Hi, @dcharkes! Do you mean that if you might use FFI for, say, mobile, but non-FFI for, say, web, that you should implement this as federated plugin? I do want to tweak the wording, but I am not sure what you mean.

Reply:
One can implement mix and match (non-FFI e.g. method channel) plugins with FFI plugins for the native platforms.

See https://github.com/dcharkes/flutter/blob/91ef6ee700b9e281539e7734f5974bc5184894fa/packages/flutter_tools/lib/src/plugins.dart#L45-L63